osx/KA-Lite/KA-Lite/AppDelegate.m

```diff
@@ -712,30 +722,32 @@ BOOL setEnvVars() {
         return FALSE;
     }
     
-    
     // Path of the KALITE_PYTHON environment variable
     NSString* envKalitePythonStr = getEnvVar(@"KALITE_PYTHON");
     if (! pathExists(envKalitePythonStr)) {
         return FALSE;
     }
+    
     NSString *KaliteHomeStr = [NSString stringWithFormat:@"%@",
                                [NSString stringWithFormat:@"launchctl setenv KALITE_HOME \"%@\"", kaliteHomePath]
                                ];
-    NSString *KalitePythonStr = [NSString stringWithFormat:@"%@",
-                                 [NSString stringWithFormat:@"launchctl setenv KALITE_PYTHON \"%@\"", envKalitePythonStr]
-                                 ];
     
     NSString *org = @"org.learningequality.kalite";
     NSString *target = [NSString stringWithFormat:@"%@/Library/LaunchAgents/%@.plist", NSHomeDirectory(), org];
     NSMutableDictionary *plistDict = [[NSMutableDictionary alloc] init];
     [plistDict setObject:org forKey:@"Label"];
     
-    
-    NSString *launchStr = [NSString stringWithFormat:@"%@ ; %@", KalitePythonStr, KaliteHomeStr];
+    // Append KA Lite app path if autoStartOnLogin value is TRUE.
+    NSString *launchStr = [NSString stringWithFormat:@"%@", KaliteHomeStr];
+    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"autoStartOnLogin"]){
+        NSString *kaliteAppPath = [NSString stringWithFormat:@"%@%@", [[NSBundle mainBundle] bundlePath], @"/Contents/MacOS/KA-Lite"];
+         launchStr = [NSString stringWithFormat:@"%@ ; %@", KaliteHomeStr, kaliteAppPath];
+    }
+   
     NSArray *arr = @[@"sh", @"-c", launchStr];
     [plistDict setObject:arr forKey:@"ProgramArguments"];
     [plistDict setObject:[NSNumber numberWithBool:TRUE] forKey:@"RunAtLoad"];
-    showNotification([NSString stringWithFormat:@"Setting KALITE_HOME and KALITE_PYTHON environment variables... %@", plistDict]);
+    showNotification([NSString stringWithFormat:@"Setting KALITE_HOME environment variables... %@", plistDict]);
     
     // Override org.learningequality.kalite.plist content
     BOOL ret = [plistDict writeToFile:target atomically:YES];
```

Comment:
Can we have an exception handler here?  Unless the `writeToFile` method handles it already and will always return a `BOOL`?